src/train_model.py

```diff
@@ -598,11 +639,26 @@ def eval_net(network, test_set, test_loader, device, criterion, dataset, _run):
                     data["output"] == predicted).all(-1).long().sum().item()
                 loss_sum += loss.item()
                 num_batches += 1
-
+            else:
+                inputs, labels = data[0].to(device), data[1].to(device)
+                outputs = network(inputs)
+                _, predicted = torch.max(outputs.data, 1)
+                loss = criterion(outputs, labels)
+                total += labels.size(0)
+                if dataset != "simple_dataset":
```

Comment:
could nested if statement be turned to if... elif... else?

Reply:
Hmm, to me that feels uglier? There's 7 lines of code that's shared, and 1 line that's conditional. So to convert to elif/else, we'd need to duplicate those 7 lines, and maintain any future edits as applying to both

Reply:
makes sense